### PR TITLE
move netdev interface to use net/netip for IP addr/ports

### PR DESCRIPTION
**Pull request by scottfeldman**
Matches changes to netdever interface made in tinygo-org/net#8.  These two PRs are paired.  If one goes, both go.

### CODE REVIEW

File: espat/espat.go [hunk 1]
```diff
@@ -44,8 +45,8 @@ type Config struct {
 type socket struct {
 	inUse    bool
 	protocol int
-	lip      net.IP
-	lport    int
+	lip      netip.Addr
+	lport    uint16
```
Comment (soypat) [inline espat/espat.go]: could be paired as a netip.AddrPort as `laddr`

**Review by soypat — APPROVED** [review]
LGTM!

File: espat/espat.go (hunk 1 above)
Comment (scottfeldman) [reply espat/espat.go]: fixed

**Comment by deadprogram** [comment]
Now merging, thanks for working on this @scottfeldman and for review @soypat

**Comment by soypat** [comment]
Was looking at the code again and just thought to myself this was a really good move! Lots of heap allocations seem to be gone now! Thanks @scottfeldman for the contribution!

**Comment by scottfeldman** [comment]
Yes, it was worth it.  What next @soypat?

**Comment by soypat** [comment]
@scottfeldman Now we use the API! 
As for the netdev-3 PR I'm still working on the TCP implementation. it's coming along nicely but its got a while to go before being lwip!

**Pull request merged**